roles/puppet_repositories/tasks/debian.yml

```
@@ -1,10 +1,16 @@
 ---
-- name: 'Install GPG package'
+- name: Install GPG package
   package:
-    name: 'gpg'
+    name: gpg
     state: present
     update_cache: true
 
-- name: 'Install Puppet repository'
+- name: Install Puppet repository
   apt:
-    deb: https://apt.puppet.com/puppet{{ foreman_puppet_repositories_version }}-release-{{ ansible_distribution_release }}.deb
+    deb: "https://apt.puppet.com/puppet{{ foreman_puppet_repositories_version }}-release-{{ ansible_distribution_release }}.deb"
+    state: present
+  register: puppet_apt_deb_install
+
+- name: Update apt cache if necessary
+  apt: update_cache=yes force_apt_get=yes
```

Please prefer the nested yaml style than inline approach for declaring parameters:

```suggestion
  apt:
    update_cache: yes
    force_apt_get: yes
```

---

Agreed! Updated.